core/src/main/java/com/linecorp/armeria/common/logging/DefaultRequestLog.java

```diff
@@ -732,6 +732,36 @@ public Throwable requestCause() {
         return requestCause;
     }
 
+    @Override
+    public void requestCause(Throwable cause) {
+        if (isAvailable(RequestLogProperty.REQUEST_CAUSE)) {
+            return;
+        }
+
+        requireNonNull(cause, "cause");
+        setRequestCause(cause, true);
```

Comment:
Shouldn't we always update the flag?
```suggestion
        setRequestCause(cause);
        updateFlags(RequestLogProperty.REQUEST_CAUSE);
```